### PR TITLE
requirements.txt updates

### PR DESCRIPTION
**Pull request by deb-intel**
sphinxcontrib-applehelp was recently updated to 1.0.6. This caused a series of compatibility issues between sphinxcontrib-applehelp, sphinx, and breathe, which caused all recent builds to fail. To lessen this problem both now and in the future, we removed hardcoded software versions in the requirements.tx file although we updated sphinx to 7.0 or higher. This will enable the system to pick compatible software versions to match our needs.

### CODE REVIEW

**Comment by mwasko** [comment]
> sphinxcontrib-applehelp was recently updated to 1.0.6. This caused a series of compatibility issues between sphinxcontrib-applehelp, sphinx, and breathe, which caused all recent builds to fail.

It looks like a gap in the requirements list to point to specific sphinxcontrib-applehelp version?

> To lessen this problem both now and in the future, we removed hardcoded software versions in the requirements.tx file although we updated sphinx to 7.0 or higher. This will enable the system to pick compatible software versions to match our needs.

How this approach is going to guarantee that our SOF documentation will compile successfully with latest tools version?

**Comment by lgirdwood** [comment]
> > sphinxcontrib-applehelp was recently updated to 1.0.6. This caused a series of compatibility issues between sphinxcontrib-applehelp, sphinx, and breathe, which caused all recent builds to fail.
> 
> It looks like a gap in the requirements list to point to specific sphinxcontrib-applehelp version?
> 
> > To lessen this problem both now and in the future, we removed hardcoded software versions in the requirements.tx file although we updated sphinx to 7.0 or higher. This will enable the system to pick compatible software versions to match our needs.
> 
> How this approach is going to guarantee that our SOF documentation will compile successfully with latest tools version?

IIRC, we align with the same versions used by the Zephyr docs (as SOF docs and Zephyr docs use the same framework). @deb-intel is my understanding still correct ?

**Comment by deb-intel** [comment]
@mwasko @lgirdwood When @intelkevinputnam and I built SOF locally, the build set the proper requirements based on our setting Sphinx to >=7. It automatically new to use sphinxcontrib-applehelp version 1.0.4 instead of 1.0.6 as well as the best version of breathe. The build kept failing because we kept giving it specific versions. I think this will continue to happen every time breathe, docutils, and software we didn't even know about (such as -applehelp) makes updates.

**Comment by deb-intel** [comment]
IIRC, we align with the same versions used by the Zephyr docs (as SOF docs and Zephyr docs use the same framework). @deb-intel is my understanding still correct ?

@lgirdwood I am not familiar with how Zephyr handles things. I know they no longer have a full-time doc person. I'll ask one of our guys if he can see what's up with their requirements.

**Comment by intelkevinputnam** [comment]
I recommend moving to version 7 of sphinx. A good option for a specific version is 7.1.2 as this will have the broadest compatibility for those who might want to build it locally (later versions require Python 10 or better).

**Comment by lgirdwood** [comment]
Thanks @deb-intel @intelkevinputnam for explaining.

**Comment by lgirdwood** [comment]
@mwasko good for you now ?

**Review by mwasko — APPROVED** [review]

**Comment by marc-hb** [comment]
> When @intelkevinputnam and I built SOF locally,

What is "locally"? Ubuntu, Fedora, other? Which version? Each distribution and even each user has a different combination. Versions were hardcoded in `requirements.txt` to solve that problem (We already have `requirements-lax.txt` for more flexibility)

> The build kept failing because we kept giving it specific versions.

In that case you just un-hardcode locally  and temporarily, find a complete set of versions that works (as you already did) and then snapshot and hardcode that combination. Problem solved for everyone and for a long time as opposed to just on one system for a short time. That has been working and it can work again.

> It looks like a gap in the requirements list to point to specific sphinxcontrib-applehelp version?

Yes of course. The solution was to hardcode MORE, not less.

> I think this will continue to happen every time breathe, docutils, and software we didn't even know about (such as -applehelp) makes updates.

It will for sure happen again very soon now that all the versions are un-harcode. In fact it's already happening:

- Using `requirements-lax.txt` I just hit this recent regression on my system: https://github.com/sphinx-doc/sphinx/issues/9433 People have been avoiding it by... hardcoding.

- `blockdiag` is now orphaned (#472). We have been (and still are!) avoiding that failure by hardcoding pillow (73ac8f55622d)

- etc.

PS: I spent days finding combinations that work, please copy me next time.

File: scripts/requirements.txt [hunk 1]
```diff
@@ -1,11 +1,13 @@
 # This file hardcodes validated versions with '==',
 # see requirements-lax.txt for an alternative.
-breathe==4.29.2
-sphinx==4.5.0
-docutils==0.17.1
+
```
Comment (marc-hb) [inline scripts/requirements.txt:3]: The comment still says "This file hardcodes..."

**Pull request merged**